listenbrainz/webserver/views/api_compat.py

```diff
@@ -79,6 +80,7 @@ def api_auth_approve():
 
 @api_bp.route('/2.0/', methods=['POST', 'GET'])
 @ratelimit()
+@api_listenstore_needed
```

Comment:
future improvement - some methods in this endpoint need ts and some don't. If we plan to support this API into the future then we should allow submissions to work.

This endpoint returns XML on failure not json

Reply:
On the some bits vs others -- yeah, I saw this, but the API sees little use, so I think its ok for now.

Reply:
I updated this to only return an error message on user.getInfo (using the lastfm error messages), although this is broken anyway: LB-877